### ISSUE TITLE
running concurrent tests with multiple CeleryClient and multiple tasks fails

### ISSUE THREAD

**Issue by shicky** [shicky/gocelery#21]
As apparent from below log, `TestWorkerClientArgs` test is attempting to get registeredTasks map `0xc4200e48d0` instead of `0xc4200e5590` which is from `TestWorkerClientKwargs` test.
- both tests have new CeleryClient of its own: `0xc4200e4900`, `0xc4200e55c0`
- both tests pass if run individually.

```
=== RUN   TestWorkerClientKwargs
2016/09/30 20:49:41 kwarg client 0xc4200e4900
2016/09/30 20:49:41 registering task on 0xc4200e48d0 multiply_kwargs
2016/09/30 20:49:41 registered tasks: 0xc4200e48d0 map[multiply_kwargs:0xc4200fabc0]
2016/09/30 20:49:41 getting tasks: 0xc4200e48d0 map[multiply_kwargs:0xc4200fabc0]
2016/09/30 20:49:41 kwarg client 0xc4200a1b00
2016/09/30 20:49:41 registering task on 0xc4200a1ad0 multiply_kwargs
2016/09/30 20:49:41 registered tasks: 0xc4200a1ad0 map[multiply_kwargs:0xc4200fadc0]
2016/09/30 20:49:41 getting tasks: 0xc4200a1ad0 map[multiply_kwargs:0xc4200fadc0]
--- PASS: TestWorkerClientKwargs (0.02s)
=== RUN   TestWorkerClientArgs
2016/09/30 20:49:41 arg client 0xc4200e55c0
2016/09/30 20:49:41 registering task on 0xc4200e5590 multiply
2016/09/30 20:49:41 registered tasks: 0xc4200e5590 map[multiply:0x487510]
2016/09/30 20:49:41 getting tasks: 0xc4200e48d0 map[multiply_kwargs:0xc4200fabc0]
2016/09/30 20:49:41 task multiply_args is not registered
2016/09/30 20:49:41 async result got
--- FAIL: TestWorkerClientArgs (5.01s)
    gocelery_test.go:134: failed to get result: 5s timeout getting result for cdd00c82-e593-45a0-955a-4ac10f2b4b09
```

**Comment by shicky** [comment]
In current implementation, when `StopWorker()` is called, `gocelery` does not wait for all workers to exit.
This causes older worker processing newer requests causing above error.

There is a bug where `w.workerWG.Wait()` is called inside `StartWorker()` function instead of `StopWorker()` function.

Related Issue:
`gocelery` currently can support only one client per machine - refer to #23

**Comment by shicky** [comment]
`GetTaskMessage` should not be blocking.
There is a bug in `amqp_broker.go` where GetTaskMessage() blocks on `consumingChannel`.

**Comment by shicky** [comment]
streadway/amqp does NOT allow concurrent use of channels.
`TestWorkerClient*` tests should be merged and not run in parallel.

[https://github.com/streadway/amqp/issues/170](https://github.com/streadway/amqp/issues/170)

**Comment by shicky** [comment]
Fix in place 2544d6720d82fd6ac955137722d3c1f2ccf05027
* waiting for all workers to finish on `StopWorker()`
* concurrent tests removed due to limitation with `steadway/amqp` library

**Comment by shicky** [comment]
Following statement must be reviewed:
* `GetTaskMessage()` of CeleryBroker and `GetResult()` of CeleryBackend must be non-blocking.
